### PR TITLE
[JENKINS-54772] Fix reported list of environment variables from agent

### PR DESCRIPTION
**Pull request by pjanouse**
See [JENKINS-54772](https://issues.jenkins-ci.org/browse/JENKINS-54772).

### Proposed changelog entries

* JENKINS-54772, Expose real full env vars map from agent for UI presentation

### Submitter checklist

- [x] JIRA issue is well described
- [x] Changelog entry appropriate for the audience affected by the change (users or developer, depending on the change). [Examples](https://github.com/jenkins-infra/jenkins.io/blob/master/content/_data/changelogs/weekly.yml)
      * Use the `Internal: ` prefix if the change has no user-visible impact (API, test frameworks, etc.)
- [ ] Appropriate autotests or explanation to why this change has no tests
- [ ] For dependency updates: links to external changelogs and, if possible, full diffs

### Desired reviewers

@mention

### CODE REVIEW

**Review by oleg-nenashev — CHANGES_REQUESTED** [review]
IMHO it needs a security review before merging, because sensitive system environment variables are being returned back to the master. The page with `systemInfo` visualization is visible to users with `Computer.CONNECT` only, and probably it's enough

CC @daniel-beck

**Comment by pjanouse** [comment]
@oleg-nenashev Sure, on the other hand, `EnvVars` doesn't contain any filtering yet.

**Comment by pjanouse** [comment]
Created https://issues.jenkins-ci.org/browse/SECURITY-1304 for tracking security related  effort.

File: core/src/main/java/hudson/model/Computer.java
Comment (jvz) [inline core/src/main/java/hudson/model/Computer.java]:
```suggestion
            Map<String,String> env = new TreeMap<String,String> ();
```

File: core/src/main/java/hudson/model/Computer.java
Comment (jvz) [inline core/src/main/java/hudson/model/Computer.java]:
```suggestion
            Map<String, String> env = new TreeMap<>(System.getenv());
```

**Comment by varyvol** [comment]
@oleg-nenashev @daniel-beck could you please review?

File: core/src/main/java/hudson/model/Computer.java
Comment (Wadeck) [inline core/src/main/java/hudson/model/Computer.java]: FTR nothing new here, just copy/paste from [EnvVars](https://github.com/jenkinsci/jenkins/blob/004b23133f2344c909c67d04e8f6d860fa03e5ad/core/src/main/java/hudson/EnvVars.java#L451-L459)

File: core/src/main/java/hudson/model/Computer.java
Comment (Wadeck) [inline core/src/main/java/hudson/model/Computer.java]: FTR nothing new here, just copy/paste from [EnvVars](https://github.com/jenkinsci/jenkins/blob/004b23133f2344c909c67d04e8f6d860fa03e5ad/core/src/main/java/hudson/EnvVars.java#L426)

File: core/src/main/resources/jenkins/slaves/systemInfo/EnvVarsSlaveInfo/systemInfo.jelly
Comment (Wadeck) [inline core/src/main/resources/jenkins/slaves/systemInfo/EnvVarsSlaveInfo/systemInfo.jelly]: FTR the `it` refers to the `SlaveComputer` in this case, it's the `instance` that refer to the `SlaveSystemInfo`.

File: core/src/main/java/hudson/model/Computer.java
Comment (Wadeck) [inline core/src/main/java/hudson/model/Computer.java]:
```suggestion
            Map<String, String> env = new TreeMap<>(System.getenv());
```

To avoid an "unchecked assignment"

**Review by Wadeck — CHANGES_REQUESTED** [review]
From functional PoV, the idea is good. After my tests 
1) on Windows the env vars name are case insensitive, so, no real impact, 
2) running a linux docker from master providing env vars using `-e` shows me the expected behavior, **so that works** and
3) the env variables configured in the node are not included in the "regular" env variables (and they are case insensitive)

<details>

<summary>screenshot docker/unix</summary>

Screenshot_2019-04-11_104238_001

</details>


The only thing that I would like to see changed is the location of the new code. As it only impacts the `SlaveComputer`, I would suggest to move the code this class instead of the parent (`Computer`).
If you really want to put the code in `Computer`, please move your new methods after the `getEnvironment()` method to avoid breaking easy readability between `getEnvVars` and `getEnvironment`.

**Comment by oleg-nenashev** [comment]
@pjanouse are you still interested in it?

**Comment by pjanouse** [comment]
@oleg-nenashev Yes, I'll work on changes Wadeck suggested.

File: core/src/main/resources/jenkins/slaves/systemInfo/EnvVarsSlaveInfo/systemInfo.jelly
Comment (pjanouse) [reply core/src/main/resources/jenkins/slaves/systemInfo/EnvVarsSlaveInfo/systemInfo.jelly]: Done

**Comment by pjanouse** [comment]
Guys, requested changes were incorporated.

**Review by varyvol — APPROVED** [review]

**Review by oleg-nenashev — APPROVED** [review]
Okay with me. Note that synchronous calls to agents are generally a bad idea. Maybe it makes sense to add async loading of the table.

**Review by Wadeck — APPROVED** [review]
My requested change was integrated, thank you.

I did not attempt to break it.

Do we want to wait for [SECURITY-1304](https://issues.jenkins-ci.org/browse/SECURITY-1304) ? (cc @daniel-beck)

**Comment by oleg-nenashev** [comment]
@Wadeck @daniel-beck is it still on hold?

**Comment by Wadeck** [comment]
@oleg-nenashev I will let @daniel-beck decides on that ticket. We have not prioritized any work on that particular ticket in the CERT backlog.

**Comment by oleg-nenashev** [comment]
retriggering CI

**Comment by oleg-nenashev** [comment]
I am about taking silence as acceptance. If there is no negative feedback, I will merge it tomorrow

**Pull request merged**
